insights/parsers/mount.py

```diff
@@ -10,30 +10,40 @@
     /dev/mapper/HostVG-Config on /etc/shadow type ext4 (rw,noatime,seclabel,stripe=256,data=ordered)
     dev/sr0 on /run/media/root/VMware Tools type iso9660 (ro,nosuid,nodev,relatime,uid=0,gid=0,iocharset=utf8,mode=0400,dmode=0500,uhelper=udisks2) [VMware Tools]
 
-The information is stored as a list of ``AttributeDict`` objects.  Each
-``AttributeDict`` object contains attributes for the following information that
-are listed in the same order as in the command output::
-
-    - filesystem: (str) Name of filesystem
-    - mount_point: (str) Name of mount point for filesystem
-    - mount_type: (str) Name of filesystem type
-    - mount_options: (AttributeDict) Mount options as ``AttributeDict`` object
-    - mount_label: (str) Only present if optional label is present
-    - mount_clause: (str) Full string from command output
-
-The `` mount_options`` contains the mount options as attributes accessible
-via the attribute name as it appears in the command output.  For instance, the
-options ``(rw,dmode=0500)`` may be accessed as ''mnt_row_info.rw`` with the value
-``True`` and ``mnt_row_info.dmode`` with the value "0500".  The ``in`` operator
-may be used to determine if an option is present.
+The information is stored as a list of :class:`MountEntry` objects.  Each
+:class:`MountEntry` object contains attributes for the following information that
+are listed in the same order as in the command output:
+
+ * ``filesystem`` - Name of filesystem
+ * ``mount_point`` - Name of mount point for filesystem
+ * ``mount_type`` - Name of filesystem type
+ * ``mount_options`` - Mount options as a dictionary
+ * ``mount_label`` - Only present if optional label is present
+ * ``mount_clause`` - Full string from command output
+
+The ``mount_options`` is wrapped as a :class:`MountOpts` object which contains
+below fixed attributes:
+
+* ``rw`` - Read write
+* ``ro`` - Read only
+* ``defaults`` - Use default options: rw, suid, dev, exec, auto, nouser, async, and relatime
+* ``relatime`` - Inode access times relative to modify or change time
+* ``seclabel`` - `seclabel` is enabled or not
+* ``attr2`` - `opportunistic` improvement is enabled or not
+* ``inode64`` - `inode64` is enabled or not
+* ``noquota`` - Disk quotas are enforced or not
+
+For instance, the option ``rw`` in ``(rw,dmode=0500)`` may be accessed as
+``mnt_row_info.rw`` with the value ``True``, but the ``dmode`` can only be
+accessed as ``mnt_row_info.get('dmode')`` or ``mnt_row_info['dmode']`` with the
+value ``0500``.
```

Comment:
@xiangce why wouldn't you just add `dmode` (and other non-fixed attributes) to the attributes instead of changing to dictionary access, to maintain compatibility with the previous version?

Reply:
@bfahr - just wanted to only set those explicitly documented `fixed attributes` as attributes (try to fix the issue that Mike raised in email), so I didn't add these options no in the `fixed attributes`, such as `dmode`, to attributes.

Now I fixed it as all keys in `data` and `fixed attributes` will be added as attributes.